plugins/PostProcessingPlugin/scripts/PurgeLinesAndUnload.py

```diff
@@ -0,0 +1,752 @@
+            midpoint_y = machine_depth / 2
+            max_x = machine_width - 7
+            min_x = 7
+            max_y = machine_depth - 7
+            min_y = 7
+        elif origin_at_center:
+            midpoint_x = 0
+            midpoint_y = 0
+            max_x = (machine_width / 2) - 7
+            min_x = -abs((machine_width / 2) - 7)
+            max_y = (machine_depth / 2) - 7
+            min_y = -abs((machine_depth / 2) - 7)
+        if purge_end_loc == "LF":
+            if prime_tower_loc == "LF":
+                adj_lines = ""
+            if prime_tower_loc == "RF":
+                adj_lines = f"G0 F{travel_speed} X{max_x} ; Move to edge\nG0 F600 Z0 ; nail down the string\nG0 F600 Z2 ; move up\n"
+            if prime_tower_loc == "RR":
+                adj_lines = f"G0 F{travel_speed} X{max_x} ; Move to edge\nG0 F600 Z0 ; nail down the string\nG0 F600 Z2 ; move up\n"
+            if prime_tower_loc == "LR":
+                adj_lines = f"G0 F{travel_speed} Y{max_y} ; Move to edge\nG0 F600 Z0 ; nail down the string\nG0 F600 Z2 ; move up\n"
+        elif purge_end_loc == "RR":
+            if prime_tower_loc == "LF":
+                adj_lines = f"G0 F{travel_speed} X{min_x} ; Move to edge\nG0 F600 Z0 ; nail down the string\nG0 F600 Z2 ; move up\n"
+            if prime_tower_loc == "RF":
+                adj_lines = f"G0 F{travel_speed} Y{min_y} ; Move to edge\nG0 F600 Z0 ; nail down the string\nG0 F600 Z2 ; move up\n"
+            if prime_tower_loc == "RR":
+                adj_lines = ""
+            if prime_tower_loc == "LR":
+                adj_lines = f"G0 F{travel_speed} X{min_x} ; Move to edge\nG0 F600 Z0 ; nail down the string\nG0 F600 Z2 ; move up\n"
+        return adj_lines
+
+    # Determine the quadrant that the prime tower rests in so the adjustments can be calculated
+    def _prime_tower_quadrant(self, prime_tower_x, prime_tower_y, bed_shape, origin_at_center, machine_width, machine_depth):
+        if not origin_at_center:
+            midpoint_x = machine_width / 2
+            midpoint_y = machine_depth / 2
+            max_x = machine_width - 7
+            min_x = 7
+            max_y = machine_depth - 7
+            min_y = 7
+        elif origin_at_center:
+            midpoint_x = 0
+            midpoint_y = 0
+            max_x = (machine_width / 2) - 7
+            min_x = -abs((machine_width / 2) - 7)
+            max_y = (machine_depth / 2) - 7
+            min_y = -abs((machine_depth / 2) - 7)
+        if prime_tower_x < midpoint_x and prime_tower_y < midpoint_y:
+            prime_tower_location = "LF"
+        elif prime_tower_x > midpoint_x and prime_tower_y < midpoint_y:
+            prime_tower_location = "RF"
+        elif prime_tower_x > midpoint_x and prime_tower_y > midpoint_y:
+            prime_tower_location = "RR"
+        elif prime_tower_x < midpoint_x and prime_tower_y > midpoint_y:
+            prime_tower_location = "LR"
+        return prime_tower_location
```

Comment:
```suggestion
        # Determine coordinate limits based on origin position
        if origin_at_center:
            midpoint_x, midpoint_y = 0, 0
            max_x, min_x = (machine_width / 2) - 7, -((machine_width / 2) - 7)
            max_y, min_y = (machine_depth / 2) - 7, -((machine_depth / 2) - 7)
        else:
            midpoint_x, midpoint_y = machine_width / 2, machine_depth / 2
            max_x, min_x = machine_width - 7, 7
            max_y, min_y = machine_depth - 7, 7

        # Determine quadrant of the prime tower
        if prime_tower_x < midpoint_x and prime_tower_y < midpoint_y:
            return "LF"  # Lower-Front
        elif prime_tower_x > midpoint_x and prime_tower_y < midpoint_y:
            return "RF"  # Right-Front
        elif prime_tower_x > midpoint_x and prime_tower_y > midpoint_y:
            return "RR"  # Right-Rear
        elif prime_tower_x < midpoint_x and prime_tower_y > midpoint_y:
            return "LR"  # Lower-Rear
```

you can do a bit of a trick and a reverse on the conditions